WIP: roachprod: Support placement policies in AWS and GCP.

miretskiy opened the pull request:
Support node placement policies in AWS and GCP.

Release Notes: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/55529)
<!-- Reviewable:end -->

Commit:
roachprod: Make it possible to specify file descriptor limit.

Add a `--num-files-limit` flag to roachprod start command
to change the default file descriptor limit for cockroach process.

Release Notes: None
